awx/main/dispatch/publish.py

```diff
@@ -0,0 +1,116 @@
+import inspect
```

Comment:
This code implements the `@task()` decorator.  You can decorate any function or class (you'll see the associated changes in `awx.main.tasks` in this diff), and the decorator will make that class/function have an `apply_async()` and `delay()` function that match celery's interface.  In this way, most of the code that used to use celery for queueing a task hasn't changed much (this is backwards compatible with how you use celery).